src/test/java/cz/codecamp/logger/loggers/FileLoggerTest.java

```diff
@@ -0,0 +1,124 @@
+
+    @Before
+    public void setUp() throws Exception {
+        logger = new FileLogger();
+        now = LocalDateTime.now();
+        dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+    }
+
+    @Rule
+    public TemporaryFolder tempFolder = new TemporaryFolder();
+
+    @Test
+    public void testOutputOdStdOutLoggerWithUnspecifiedFormatter() throws Exception {
+        logger.setLogsFolder(tempFolder.getRoot().toPath().toString());
+
+        logger.log(LogLevelEnum.INFO, "test");
+        StackTraceElement[] ste = Thread.currentThread().getStackTrace(); // ! Must be right after calling log method -> to get proper line number in test result
+
+        String actual = getLines(logger.getLogsFolder());
+        String expected = getExpectedResult(ste[1]);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testOutputOdStdOutLoggerWithGivenFormatter() throws Exception {
+        logger.setLogsFolder(tempFolder.getRoot().toPath().toString());
+
+        FormatterInterface logFormatter = new LogFormatter();
+
+        logger.log(LogLevelEnum.INFO, "test", logFormatter);
+        StackTraceElement[] ste = Thread.currentThread().getStackTrace(); // ! Must be right after calling log method -> to get proper line number in test result
+
+        String actual = getLines(logger.getLogsFolder());
+        String expected = getExpectedResult(ste[1]);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testOutputOdStdOutLoggerWithUnspecifiedFormatterWithDayBasedFileHandler() throws Exception {
+
+        logger = new FileLogger(new DayBasedFileHandler("app.log"));
+        logger.setLogsFolder(tempFolder.getRoot().toPath().toString());
+
+        logger.log(LogLevelEnum.INFO, "test");
+        StackTraceElement[] ste = Thread.currentThread().getStackTrace(); // ! Must be right after calling log method -> to get proper line number in test result
+
+        String actual = getLines(logger.getLogsFolder());
+        String expected = getExpectedResult(ste[1]);
+        assertEquals(expected, actual);
+    }
+
+    private String getLines(String logsFolder) throws Exception {
+        File folder = new File(logsFolder);
+        File[] listOfFiles = folder.listFiles();
+        Path path = listOfFiles[0].toPath();
+        List<String> lines = Files.readAllLines(path);
+        return lines.get(0);
+    }
+
+    private String getExpectedResult(StackTraceElement ste) {
+        return "[INFO] [" + dateTimeFormatter.format(now) + "] test Class: " + ste.getClassName() + " line: " + (ste.getLineNumber()-1);
+    }
+
+    @Test
+    public void testLogFileRotation() throws Exception {
+        logger = new FileLogger(new IntervalBasedFileHandler("app.log", ChronoUnit.SECONDS, 1));
+        logger.setLogsFolder(tempFolder.getRoot().toPath().toString());
+
+        logger.log(LogLevelEnum.WARNING, "test random message");
+
+        Thread.sleep(2000);
```

Comment:
Další případ, kdy by se hodilo od konkrétního času odstínit. Teďka v jednom testu to až tak vidět není, ale pokud budeš mít v test suite 50 takových testů, kde by se mělo čekat X vteřin, aby se něco stalo, tak se učekáš - spuštění testů by mělo být co nejrychlejší.